### PR TITLE
CI to run install in multiple python versions

### PR DESCRIPTION
**Pull request by kmilo9999**
- It run pip install of the project in python 3.9,3.10 and 3.11

### CODE REVIEW

File: .github/workflows/python_ci.yml [hunk 1]
```diff
@@ -3,13 +3,16 @@ name: Run base code on different python versions
...
     steps:
+      - name: install mpi
+        run: sudo apt update && sudo apt-get install openmpi-bin openmpi-doc libopenmpi-dev 
       - uses: actions/checkout@v3
       - name: setup python
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.version }} # install the python version needed
-      - run: python install --editable ".[dev]"
+      - run: pip install --editable ".[dev]"
```
Comment (hollandjg) [inline .github/workflows/python_ci.yml:18]: This probably doesn't need to be `--editable` – as you're not going to edit it here, you can install the package in its normal mode. I'd also wonder whether the `[dev]` option is necessary – perhaps the check here could be "does the package work in the way a typical user would use it, that is without the dev dependencies?"
```suggestion
      - run: pip install .
```

File: .github/workflows/python_ci.yml [hunk 2]
```diff
@@ -3,13 +3,16 @@ name: Run base code on different python versions
...
+        version: ['3.9', '3.10', '3.11']
     runs-on: ubuntu-latest
     steps:
+      - name: install mpi
+        run: sudo apt update && sudo apt-get install openmpi-bin openmpi-doc libopenmpi-dev 
       - uses: actions/checkout@v3
       - name: setup python
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.version }} # install the python version needed
```
Comment (hollandjg) [inline .github/workflows/python_ci.yml:17]: You could make the install faster by adding cacheing:
```suggestion
          python-version: ${{ matrix.version }} # install the python version needed
          cache: "pip"
```

**Review by hollandjg — COMMENTED** [review]
Thanks for this! I've a couple of questions here, and a suggestion to do with cacheing the dependencies.

File: .github/workflows/python_ci.yml (hunk 1 above)
Comment (kmilo9999) [reply .github/workflows/python_ci.yml:18]: Good point. I'll test it without the `[dev]` option.

**Review by hollandjg — APPROVED** [review]
Great!

**Pull request merged**